phpBB/phpbb/db/driver/driver.php

```diff
@@ -793,6 +804,130 @@ function sql_build_query($query, $array)
 		return $sql;
 	}
 
+
+	protected function _process_boolean_tree_first($operations_ary)
+	{
+		// In cases where an array exists but there is no head condition,
+		// it should be because there's only 1 WHERE clause. This seems the best way to deal with it.
+		if ($operations_ary[0] !== 'AND' &&
+			$operations_ary[0] !== 'OR')
+		{
+			$operations_ary = array('AND', $operations_ary);
+		}
+		return $this->_process_boolean_tree($operations_ary) . "\n";
+	}
+
+	protected function _process_boolean_tree($operations_ary)
+	{
+		$operation = array_shift($operations_ary);
+
+		foreach ($operations_ary as &$condition)
```

Comment:
If I want to keep the algorithm, yes.
If I want to keep the final result even if I use a different algorithm, no.
Regardless, If I don't use a reference, I'll have to change the algorithm significantly. How much exactly... I'm unsure.